### PR TITLE
Allow disabling logging by setting AUSTIN_NO_LOGGING in the environment

### PR DESCRIPTION
**Pull request by Mortal**
Add a check in logging.c for the environment variable AUSTIN_NO_LOGGING. If it is nonempty, logging is disabled on all platforms.

### Alternate Designs

None considered.

### Regressions

No possible side-effects, as the odds of existing users inadvertently setting AUSTIN_NO_LOGGING are minimal.

### Verification Process

To test (on Linux):

1. Compile with `gcc -O3 -Os -Wall -pthread src/*.c -o src/austin`
2. Open a separate terminal to watch the syslog output with `journalctl -f -t austin -n0`
3. Run `AUSTIN_NO_LOGGING=1 ./src/austin python3 -c 'import time; time.sleep(1)'` and observe that nothing is output to syslog
4. Run `AUSTIN_NO_LOGGING= ./src/austin python3 -c 'import time; time.sleep(1)'` and observe the usual output to syslog
5. Run `./src/austin python3 -c 'import time; time.sleep(1)'` and observe the usual output to syslog

### CODE REVIEW

**Comment by P403n1x87** [comment]
👋 thanks for your contribution. This feature sounds interesting. I'm just curious, could you share with me your specific use case (while I figure out why the workflows are not running)? Also, it would be great if you could target the `devel` branch instead. Cheers!

**Comment by Mortal** [comment]
As for my use case, we're running several networked daemons written in Python that do a lot of marshaling/unmarshaling and database queries, and during certain events we're getting really unsatisfactory load. It would be nice if I could attach austin in our production environment and obtain some samples to indicate where to optimize. However I don't want to spew syslog in our production environments with austin's logging. If austin could log to stderr and output samples to stdout then maybe that could work for me as well, but I thought it would be easiest to just disable logging altogether.

**Comment by P403n1x87** [comment]
@Mortal thanks for sharing your use case. Since this is a new feature, and in fact the first supported environment variable, we would need to:

- add an entry in the change-log
- document the environment variable interface in the README
- reference the new variable in the Logging section of the README
- add simple fork and attach test cases with the variable set to ensure that we don't crash

As for the second entry in the list above, I would add the Environment Interface subsection under Usage in which we briefly say that some Austin behaviour can be configured via environment variables, and then add a table with the supported variables (just `AUSTIN_NO_LOGGING` in this case). Are you happy to also contribute these changes?

Also, just a heads-up that I might also ask you to rebase your branch on devel, once I push the GitHub Actions changes required to run on PRs.

**Comment by P403n1x87** [comment]
@Mortal PR #122 is now merged. Please rebase on `devel` when you can 🙏

**Comment by Mortal** [comment]
@P403n1x87 I've rebased, updated the change-log and README, and added a test per your feedback. Please take another look. Also, it seems you need to give approval to run the workflows: `First-time contributors need a maintainer to approve running workflows.`

File: test/test_fork.py
Comment (P403n1x87) [inline test/test_fork.py:245]:
```suggestion
def test_no_logging(py, tmp_path, monkeypatch):
    monkeypatch.setenv("AUSTIN_NO_LOGGING", "1") 
    result = austin("-i", "1ms", *python(py), target("target34.py"))
    assert has_pattern(result.stdout, "target34.py:keep_cpu_busy:3"), compress(
        result.stdout
    )
```
Bonus if we could also test that there are no Austin messages in logs (in a platform-specific way) :slightly_smiling_face: .

It would also be great if we could run this test as additional parameters for existing tests, but this can be addressed with a separate PR.

Comment (Mortal) [reply test/test_fork.py]: OK, I'm not so familiar with monkeypatch - my initial test function was based on a brief reading of the other tests in test_fork.py, so I'm happy that you could improve it.

File: test/utils.py
Comment (Mortal) [reply test/utils.py]: OK, reverted changes in this file (after introducing monkeypatch).

**Review by P403n1x87 — APPROVED** [review]
🙌 thanks for your contribution!

**Pull request merged**
